src/components/data-grid/header-column-toggle/HeaderColumnToggle.tsx

```diff
@@ -46,18 +44,18 @@ export const HeaderColumnToggle = ({ api: gridApi, columnApi }: HeaderColumnTogg
       const colId = column.getColId();
       const isVisible = !column?.isVisible() || false;
 
-      columnApi.setColumnVisible(colId, isVisible);
+      gridApi?.setColumnsVisible([colId], isVisible);
 
       setVisibleColumns();
     },
-    [columnApi, gridApi, setVisibleColumns],
+    [gridApi, setVisibleColumns],
   );
 
   useEffect(() => {
-    const items = columnApi
-      .getAllColumns()
-      // skip the first two columns (checkbox and first column, which is always visible)
-      ?.filter((column, index) => index > 1 && column.getColDef().headerName)
+    const items = gridApi
+      .getAllGridColumns()
+      // skip the first three columns (checkbox, rowActionItems and first column, which is always visible)
```

Comment:
If there are no `rowActionItems` does this still work?

Reply:
@hiddeottens yeah, it will work.
I just change the property hide to true if there are no rowActionItems.
But rowActionItems will be in column